File: win-prog/ProgWinVB/Toolbars and Status Bars/MenuHelpSubclass/MenuItemHelp.vb
'---------------------------------------------
' MenuItemHelp.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class MenuItemHelp
    Inherits MenuItem

    ' Private fields
    Private sbpHelpPanel As StatusBarPanel
    Private strHelpText As String

    ' Constructors
    Sub New()
    End Sub

    Sub New(ByVal strText As String)
        MyBase.New(strText)
    End Sub

    ' Properties
    Property HelpPanel() As StatusBarPanel
        Set(ByVal Value As StatusBarPanel)
            sbpHelpPanel = Value
        End Set
        Get
            Return sbpHelpPanel
        End Get
    End Property

    Property HelpText() As String
        Set(ByVal Value As String)
            strHelpText = Value
        End Set
        Get
            Return strHelpText
        End Get
    End Property

    ' Method override
    Protected Overrides Sub OnSelect(ByVal ea As EventArgs)
        MyBase.OnSelect(ea)
        If Not HelpPanel Is Nothing Then
            HelpPanel.Text = HelpText
        End If
    End Sub
End Class

File: win-prog/ProgWinCS/Toolbars and Status Bars/MenuHelpSubclass/MenuItemHelp.cs
//-------------------------------------------
// MenuItemHelp.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class MenuItemHelp: MenuItem
{
                                                  // Private fields
     StatusBarPanel sbpHelpPanel;
     string         strHelpText;
                                                  // Constructors
     public MenuItemHelp()
     {
     }
     public MenuItemHelp(string strText): base(strText)
     {
     }
                                                  // Properties
     public StatusBarPanel HelpPanel
     {
          get { return sbpHelpPanel; }
          set { sbpHelpPanel = value; }
     }
     public string HelpText
     {
          get { return strHelpText; }
          set { strHelpText = value; }
     }
                                                  // Method override
     protected override void OnSelect(EventArgs ea)
     {
          base.OnSelect(ea);

          if (HelpPanel != null)
               HelpPanel.Text = HelpText;
     }
}